node/resolvers/subscriptions/index.ts

```diff
@@ -18,5 +21,60 @@ export const queries = {
           paused: 0,
         })
     })
+  },
+
+  listSubscriptionOrdersByStatus: async (_, args, { dataSources: { subscription } }) => {
+
+    let where
+    switch (args.status) {
+      case "SUCCESSFUL":
+        where = `(status=SUCCESS OR status=SUCCESS_WITH_NO_ORDER OR status=SUCCESS_WITH_PARTIAL_ORDER) AND (date between ${args.initialDate} and ${args.endDate})`
```

Comment:
extract this string template to an aux function.

Reply:
👍